src/Symfony/Component/Messenger/TraceableMessageBus.php

```diff
@@ -65,7 +65,18 @@ private function getCaller(): array
         $file = $trace[1]['file'];
         $line = $trace[1]['line'];
 
-        for ($i = 2; $i < 8; ++$i) {
+        $handleTraitFile = (new \ReflectionClass(HandleTrait::class))->getFileName();
+        for ($i = 1; $i < 8; ++$i) {
+            if (isset($trace[$i]['file'], $trace[$i + 1]['file'], $trace[$i + 1]['line']) && $trace[$i]['file'] === $handleTraitFile) {
+                $file = $trace[$i + 1]['file'];
+                $line = $trace[$i + 1]['line'];
+                $found = true;
+
+                break;
+            }
+        }
+
+        for ($i = 2; $i < 8 && !isset($found); ++$i) {
```

Comment:
define the `$found` variable (set to `false`) before the previous loop instead of detecting undefined variable here. It makes the code more readable, and less fragile in term of changes (setting `$found = false` currently would still consider it is found)

Reply:
Alright, fixed